lib/puppet/confine/boolean.rb

```diff
@@ -0,0 +1,50 @@
+require 'puppet/confine'
+
+# Common module for the Boolean confines. It currently
+# contains just enough code to implement PUP-9336.
+class Puppet::Confine
+  module Boolean
+    # Returns the passing value for the Boolean confine. This should
+    # just be the class name.
+    def passing_value
+      return @passing_value unless @passing_value.nil?
+
+      class_name = self.class.to_s.split('::').pop.downcase.to_sym
+
+      @passing_value = class_name == :true ? true : false
```

Comment:
It seems odd to me that the mixin has knowledge about the class it's being mixed into. Rather than using mixins, what about making this a base class inheriting from `Puppet::Confine`, and having `Puppet::Confine::{True,False}` inheriting from this. Then this method would just be:

```ruby
attr_reader :passing_value
```
and then the derived class eg `Puppet::Confine::True` could be:

```ruby
def passing_value
  value = super.passing_value
  value.nil? true : value
end
```

And similarly for `Puppet::Confine::False`

Reply:
That did not work b/c of https://github.com/puppetlabs/puppet/pull/7305#discussion_r242727694.